sdk/ml/azure-ai-ml/azure/ai/ml/operations/_index_operations.py

```diff
@@ -192,3 +205,302 @@ def cls(rest_indexes: Iterable[RestIndex]) -> List[Index]:
+            )
+
+            return self.index_data(data_index=data_index, identity=input_source_credential)
+        else:
+            raise ValueError(f"Unsupported input source type {type(input_source)}")
+
+    def index_data(
+        self,
+        data_index: "azureml.rag.dataindex.DataIndex",
+        identity: Optional[Union[ManagedIdentityConfiguration, UserIdentityConfiguration]] = None,
+        compute: str = "serverless",
+        serverless_instance_type: Optional[str] = None,
+        input_data_override: Optional[Input] = None,
+        submit_job: bool = True,
+        **kwargs,
+    ) -> PipelineJob:
+        """
+        Returns the data import job that is creating the data asset.
+
+        :param data_index: DataIndex object.
+        :type data_index: azure.ai.ml.entities._dataindex
+        :param identity: Identity configuration for the job.
+        :type identity: Optional[Union[ManagedIdentityConfiguration, UserIdentityConfiguration]]
+        :param compute: The compute target to use for the job. Default: "serverless".
+        :type compute: str
+        :param serverless_instance_type: The instance type to use for serverless compute.
+        :type serverless_instance_type: Optional[str]
+        :param input_data_override: Input data override for the job.
+            Used to pipe output of step into DataIndex Job in a pipeline.
+        :type input_data_override: Optional[Input]
+        :param submit_job: Whether to submit the job to the service. Default: True.
+        :type submit_job: bool
+        :return: data import job object.
+        :rtype: ~azure.ai.ml.entities.PipelineJob.
+        """
+        from azure.ai.ml import MLClient
+        from azure.ai.ml.constants._common import AssetTypes
+        from azure.ai.ml._utils._asset_utils import (
+            _validate_auto_delete_setting_in_data_output,
+            _validate_workspace_managed_datastore,
+        )
+        from azureml.rag.dataindex import index_data as index_data_func
+
+        default_name = "data_index_" + data_index.name
+        experiment_name = kwargs.pop("experiment_name", None) or default_name
+        data_index.type = AssetTypes.URI_FOLDER
+
+        # avoid specifying auto_delete_setting in job output now
+        _validate_auto_delete_setting_in_data_output(data_index.auto_delete_setting)
+
+        # block customer specified path on managed datastore
+        data_index.path = _validate_workspace_managed_datastore(data_index.path)
+
+        # TODO: This is import_data behavior, not sure if it should be default for index_data, or just be documented?
+        if "${{name}}" not in data_index.path and "{name}" not in data_index.path:
+            data_index.path = data_index.path.rstrip("/") + "/${{name}}"
+
+        index_job = index_data_func(
+            description=data_index.description or kwargs.pop("description", None) or default_name,
+            name=data_index.name or kwargs.pop("name", None),
```

Comment:
Why do we need to remove name, description, etc. from **kwargs?

Reply:
Only"incremental_update" in **kwargs is used later. Safe to pop "description" and "name" here